cloud-watch-to-slack-testing-58006136-c30b-43d1-880c-7b36f860e7c8/deployment/index.js

```diff
@@ -142,10 +146,40 @@ function sendMessageToSlack(messageText, callback) {
   });
 }
 
+// Set the Slack channel based on the input SNS Topic to Slack Channel map
+// in the snsTopicToSlackChannel env var with format
+// {"<SNS Topic resource id>":"<slack channel name>"}
+// E.g.
+//     {"slack-low-priority-topic":"dockstore-testing",
+//     "slack-medium-priority-topic":"dockstore-dev-alerts",
+//     "slack-high-priority-topic":"dockstore-alerts",
+//     "SSM-to-Slack-SNSTopicToSlack-1VI4KZW1DSADS":"dockstore-dev-testing" }
+// input: SNS topic ARN
+function setSlackChannelBasedOnSNSTopic(topicArn) {
+  // Get the SNS Topic resource ID from the AWS ARN
+  // https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
+  const snsTopicResourceID = topicArn.slice(topicArn.lastIndexOf(":") + 1);
+  const snsTopicToSlackChannel = JSON.parse(`${snsTopicToSlackChannelJSON}`);
+
+  if (Object.keys(snsTopicToSlackChannel).includes(snsTopicResourceID)) {
+    slackChannel = snsTopicToSlackChannel[snsTopicResourceID];
```

Comment:
I don't think you should change a global variable like this. This may leak into the next invocation. Googled briefly, and here's [one link](https://blog.scottlogic.com/2019/03/04/lambda-global-state.html).

Ex:

1. Invocation 1 sets it to `dockstore-alerts`
2. In invocation 2, the test on line 164 doesn't pass (snsTopicResourceID isn't a key for some reason -- is that possible?).
3. You want it to go to the default value of `slackChannel`, but `slackChannel` is set to `dockstore-alerts` from the previous invocation, so it will go to `dockstore-alerts`.

Reply:
I think I fixed this. Please advise.